### ISSUE TITLE
Oblige Projects to have an architecture.md file

### ISSUE THREAD

**Issue by amihaiemil** [zerocracy/farm#666]
I think each project managed by ZC (including this one) should have an ``architecture.md`` file which should give a clear overview of the codebase's architecture. Something like [this](https://github.com/opencharles/charles-rest/blob/master/architecture.md) 

This file should explain:

1. where the entry point is (where to start looking into if you are new on the project) and the main code execution flow;
2. what are the main Types in the project (the main objects), how do they work together;
3. Other info relevant to understanding the code, interfaces, overall design pattern etc
4. Build plugins and tools used; what exactly happens when running ``mvn clean install``? 
5. Libraries used

This project has a little explanation in the README, it lays out the main parts, but doesn't tell how do they work together, what is called and when, why do we have dozens of groovy scripts instead of having Java classes with proper unit tests etc. 

This requirement would drastically reduce the time new developers would have to spend to learn the project.

**Comment by 0crat** [comment]
@yegor256[/z](https://www.0crat.com/u/yegor256) please, pay attention to this issue

**Comment by yegor256** [comment]
@amihaiemil it's not a school, remember? We should not write documentation just because we want our project to be properly documented. We write it only when someone is blaming us for not having it. Up to now, there were not so many requests from programmers to document anything in this project. When they show up, we document. Also, Zerocrat is not an architect bot, it's a management bot. What happens in the project technically -- is not of our concern. We can't request any project to have any files, or so. It's out of scope of this product, for now...

**Comment by amihaiemil** [comment]
@yegor256 True, it's not a school. But this is a minimum effort to help the project a lot, and it's done only once, in the beginning :) 

And whether it is a technical or management issue is very debatable. If the project has this minimum technical documentation then the management becomes easier; programmers understand better how to split their work, they deliver faster etc.
